src/core/middleware.js

```diff
@@ -45,6 +50,19 @@ export function prefixMiddleWare(req, res, next, { _config = config } = {}) {
     // We skip prepending an app if we'd previously prepended a lang and the
     // 2nd part of the URL is now a valid app.
     log.info('Application in URL is valid following prepending a lang.');
+  } else if (
+    hasValidUrlException || isValidUrlException(URLParts[1], { _config })
```

Comment:
if I remove `hasValidUrlException ||`, the tests all pass. Can you add test coverage for this or remove it? Or is that just a side effect of checking `URLParts[1]`?